### PR TITLE
Complete AndroidX support

### PR DESCRIPTION
**Pull request by arturogutierrez**
## Description
This PR removes all references to the old previous Android support libraries

## Motivation and Context
I was trying to remove Jetifier for my project (because I already migrated to AndroidX and also all third party libraries) but I couldn't because Glide was still having reference to `android.support.annotations`.
This PR removes all references to it so we don't need Jetifier to use Glide now.

I've already tested it publishing to MavenLocal and the new version is not having any dependency on Android Support.

### CODE REVIEW

**Comment by arturogutierrez** [comment]
@googlebot I signed it!

**Comment by sjudd** [comment]
I don't think any of these changes are actually necessary? ProcessorUtil references the old types by reflection, but it's only used if the dependency is already present. 

Does volley or dagger reference the support library?

**Comment by arturogutierrez** [comment]
Yep, seems are not necessary indeed. Just cleaning some code then (Volley bump was required by a lint rule).

File: annotation/compiler/src/main/java/com/bumptech/glide/annotation/compiler/ProcessorUtil.java [hunk 1]
```diff
@@ -48,7 +48,6 @@
 import javax.lang.model.type.TypeMirror;
 import javax.lang.model.type.TypeVariable;
 import javax.lang.model.util.ElementFilter;
-import javax.lang.model.util.Elements;
```
Comment (sjudd) [inline annotation/compiler/src/main/java/com/bumptech/glide/annotation/compiler/ProcessorUtil.java]: The lines removed in this file are still needed for projects that use the support library, including internal projects.

**Comment by sjudd** [comment]
I think we're not left with much other than the version bumps in the gradle files. If you'd like to submit those, I'm happy to accept.

**Comment by arturogutierrez** [comment]
@sjudd You're right. This is unnecessary indeed so can be closed

**Pull request closed without merging**